module/VuFind/src/VuFind/Controller/HoldsController.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Holds Controller
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2010.
+ * Copyright (C) The National Library of Finland 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\Controller;
+
+use Laminas\ServiceManager\ServiceLocatorInterface;
+
+/**
+ * Controller for the user holds area.
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class HoldsController extends AbstractBase
+{
+    /**
+     * Hold update results container
+     *
+     * @var \Laminas\Session\Container
+     */
+    protected $updateResultsContainer;
+
+    /**
+     * Constructor
+     *
+     * @param ServiceLocatorInterface    $sm  Service locator
+     * @param \Laminas\Session\Container $urc A session container for hold update
+     * results
+     */
+    public function __construct(ServiceLocatorInterface $sm,
+        \Laminas\Session\Container $urc
+    ) {
+        $this->serviceLocator = $sm;
```

Comment:
I'd be inclined to change this to `parent::__construct($sm);` -- since we don't use the serviceLocator directly in this class at all, I think it's more clear to show that it is a dependency of the parent class... and when we eventually refactor that, the relationship will be more clear.